pyfr/plugins/tavg.py

```diff
@@ -358,3 +363,200 @@ def __call__(self, intg):
+            amap = [i for n, i in sorted(zip(anames, aidx))]
+
+            sidx = [i for i, v in enumerate(fields) if v.startswith('std-')]
+            snames = [fields[i].removeprefix('std-') for i in sidx]
+            smap = [i for n, i in sorted(zip(snames, sidx))]
+
+            if i == 0:
+                self.stats, self.cfg = stats, cfg
+
+                self.fields = fields
+                self.cfgsect = cfgsect
+                self.region = cfg.get(cfgsect, 'region')
+                self.uuid = file['mesh_uuid']
+
+                self.idxs  = [k for k in file if re.search(r'_idxs_p\d+$', k)]
+                self.dkeys = [k for k in file if k.startswith('tavg_') and k not
+                              in self.idxs]
+
+                self.mapping = [(amap, smap)]
+                self.anames = sorted(anames)
+                self.snames = sorted(snames)
+
+                # Initialise std max and avg registers
+                nsnames = len(snames)
+                if self.std_all:
+                    self.std_max = np.zeros(nsnames)
+                    self.std_avg = np.zeros(nsnames)
+
+                # Build fun avg expressions
+                c = cfg.items_as('constants', float)
+                self.fexprs = fexprs = []
+                for k in cfg.items(cfgsect, prefix='fun-avg-'):
+                    fexprs.append(cfg.getexpr(cfgsect, k, subs=c))
+            else:
+                self.mapping.append((amap, smap))
+
+            # Check for compatibility of files
+            if self.uuid != file['mesh_uuid']:
+                raise RuntimeError('Average files computed on different meshes')
+            if self.region != cfg.get(cfgsect, 'region'):
+                raise RuntimeError('Average files computed on different regions')
+
+            for k in cfg.items(cfgsect, prefix='avg-'):
+                if cfg.get(cfgsect, k) != self.cfg.get(self.cfgsect, k):
+                    raise RuntimeError('Different average field definitions')
+
+    def _merge_data(self, outf):
+        avg_time_inv = 1/self.avg_time
+
+        (file0, stats0, cfg0, dt0), f = self._copy_pop(self.files)
+        (amap0, smap0), mapping = self._copy_pop(self.mapping)
+
+        for key in self.dkeys:
+            # Initialise accumulators
+            data = file0[key]
+            avg_acc = data[:, amap0]*dt0
+            if self.std_all:
+                var_acc = data[:, smap0]**2*dt0
+
+            # Perform the rest of the accumulations
+            for (file, stats, cfg, dt), (amap, smap) in zip(f, mapping):
+                data = file[key]
+
+                avg_acc += data[:, amap]*dt
+                if self.std_all:
+                    var_acc += data[:, smap]**2*dt
+
+            if self.fexprs:
+                avg = avg_acc.transpose(1, 0, 2)*avg_time_inv
+                fun_avg = self._eval_fun_exprs(avg)
+                outf[key] = np.vstack((avg, fun_avg)).transpose(1, 0, 2)
```

Comment:
Can transposes (different stacking function will help, can also nudge the eval fun exprs to use a different layout).